omnigibson/scenes/scene_base.py

```diff
@@ -169,85 +204,137 @@ def object_registry_group_keys(self):
     def loaded(self):
         return self._loaded
 
+    @property
+    def idx(self):
+        """Index of this scene in the simulator. Should not change."""
+        return og.sim.scenes.index(self)
+
     @property
     def initialized(self):
         return self._initialized
 
+    @property
+    def use_floor_plane(self):
+        return self._use_floor_plane
+
+    def prebuild(self):
+        """
+        Prebuild the scene USD before loading it into the simulator. This is useful for caching the scene USD for faster
+        loading times.
+
+        Returns:
+            str: Path to the prebuilt USD file
+        """
+        # Prebuild and cache the scene USD using the objects
+        if self.scene_file not in PREBUILT_USDS:
+            # Prebuild the scene USD
+            log.info(f"Prebuilding scene file {self.scene_file}...")
+
+            # Create a new stage inside the tempdir, named after this scene's file.
+            decrypted_fd, usd_path = tempfile.mkstemp(os.path.basename(self.scene_file) + ".usd", dir=og.tempdir)
+            os.close(decrypted_fd)
+            stage = lazy.pxr.Usd.Stage.CreateNew(usd_path)
+
+            # Create the world prim and make it the default
+            world_prim = stage.DefinePrim("/World", "Xform")
+            stage.SetDefaultPrim(world_prim)
+
+            # Iterate through all objects and add them to the stage
+            for obj_name, obj in self._init_objs.items():
+                obj.prebuild(stage)
+
+            stage.Save()
+            del stage
+
+            PREBUILT_USDS[self.scene_file] = usd_path
+
+        # Copy the prebuilt USD to a new path
+        decrypted_fd, instance_usd_path = tempfile.mkstemp(os.path.basename(self.scene_file) + ".usd", dir=og.tempdir)
+        os.close(decrypted_fd)
+        shutil.copyfile(PREBUILT_USDS[self.scene_file], instance_usd_path)
+        return instance_usd_path
+
     def _load(self):
         """
         Load the scene into simulator
         The elements to load may include: floor, building, objects, etc.
         """
-        # Create collision group for fixed base objects' non root links, root links, and building structures
-        CollisionAPI.create_collision_group(col_group="fixed_base_nonroot_links", filter_self_collisions=False)
-        # Disable collision between root links of fixed base objects
-        CollisionAPI.create_collision_group(col_group="fixed_base_root_links", filter_self_collisions=True)
-        # Disable collision between building structures
-        CollisionAPI.create_collision_group(col_group="structures", filter_self_collisions=True)
```

Comment:
where did all the collision group code go?

Reply:
These are now in Simulator __init__